src/dirac_cwl_proto/transformation/__init__.py

```diff
@@ -186,6 +186,13 @@ def submit_transformation_router(transformation: TransformationSubmissionModel)
         job_model_params = _generate_job_model_parameter(input_data_dict)
         logger.info("Input data for the transformation retrieved!")
 
+    elif transformation.execution_hooks.configuration:
+        job_model_params.append(
+            JobInputModel(
+                sandbox=None, cwl=transformation.execution_hooks.configuration
+            )
+        )
+
```

Comment:
Does these lines work?
I might be wrong but I think `JobInputModel.cwl` is expected to be a CWL input, whereas the `transformation.execution_hooks.configuration` is expected to be a dict mapping an input key to a given value.
They look very similar but they are not exactly the same, and I agree this is not clear within the code 😅 

I am not sure to understand why you need them? Can you explain the issue you got leading you to this block of code?

Reply:
Note: we could potentially add a `TypedDict` for the CWL input to distinct them.